contracts/strategies/HurricaneStratForLP.sol

```diff
@@ -0,0 +1,265 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.7.0;
+
+import "../YakStrategy.sol";
+import "../interfaces/IHurricaneRouter.sol";
+import "../interfaces/IHurricaneFactory.sol";
+import "../interfaces/IHurricaneMasterChief.sol";
+import "../lib/SafeMath.sol";
+import "../lib/DexLibrary.sol";
+
+/**
+ * @notice LP strategy for HurricaneSwap
+ */
+contract HurricaneStratForLP is YakStrategy {
+
+  using SafeMath for uint;
+  using SafeMath for uint256;
+  IHurricaneMasterChief public stakingContract;
+  IHurricaneRouter public router;
+  address[] route1;
+  address[] route0;
+  uint public PID;
+
+
+  constructor(
+    string memory _name,
+    address _depositToken, 
+    address _rewardToken, 
+    address _stakingContract,
+    address _router,
+    address _timelock,
+    address[] memory _route0,
+    address[] memory _route1,
+    uint _pid,
+    uint _minTokensToReinvest,
+    uint _adminFeeBips,
+    uint _devFeeBips,
+    uint _reinvestRewardBips
+  ) {
+    name = _name;
+    depositToken = IERC20(_depositToken);
+    rewardToken = IERC20(_rewardToken);
+    stakingContract = IHurricaneMasterChief(_stakingContract);
+    PID = _pid;
+    devAddr = msg.sender;
+    router = IHurricaneRouter(_router);
+    // Perform checks on the routes
+    setRoutes(_route0, _route1);
+    setAllowances();
+    updateMinTokensToReinvest(_minTokensToReinvest);
+    updateAdminFee(_adminFeeBips);
+    updateDevFee(_devFeeBips);
+    updateReinvestReward(_reinvestRewardBips);
+    updateDepositsEnabled(true);
+    transferOwnership(_timelock);
+    
+
+    emit Reinvest(0, 0);
+  }
+
+  function setRoutes(address[] memory _route0, address[] memory _route1) external onlyOwner {
```

Comment:
this needs to be onlyDev, owner and dev could be different persons with different timelocks